### PR TITLE
Expose prop to resetHighlightValue

### PR DESCRIPTION
**Pull request by jlo1**

### CODE REVIEW

File: android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java [hunk 1]
```diff
@@ -28,13 +30,24 @@ public RNOnChartValueSelectedListener(Chart chart) {
...
             ReactContext reactContext = (ReactContext) chart.getContext();
             reactContext.getJSModule(RCTEventEmitter.class).receiveEvent(
                     chart.getId(),
                     "topSelect",
                     EntryToWritableMapUtils.convertEntryToWritableMap(entry));
+            
+            // Hack: unhighlight all values to trigger callback on re-taps
+            // Ideal fix: alter native library to support flag passed in
+            final Handler handler = new Handler();
+            Runnable r = new Runnable() {
```
Comment (drhops) [inline android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: nit: final

File: android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java [hunk 2]
```diff
@@ -28,13 +30,24 @@ public RNOnChartValueSelectedListener(Chart chart) {
...
+            final Chart chart = mWeakChart.get();
 
             ReactContext reactContext = (ReactContext) chart.getContext();
             reactContext.getJSModule(RCTEventEmitter.class).receiveEvent(
                     chart.getId(),
                     "topSelect",
                     EntryToWritableMapUtils.convertEntryToWritableMap(entry));
+            
+            // Hack: unhighlight all values to trigger callback on re-taps
+            // Ideal fix: alter native library to support flag passed in
```
Comment (drhops) [inline android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: nit: add TODO and/or tracker item?

File: android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java [hunk 3]
```diff
@@ -113,7 +113,6 @@ private void drawBitmap(ThemedReactContext context, Canvas c, ReadableMap icon)
                 p.setDither(true);
                 p.setFilterBitmap(true);
                 p.setColorFilter(new PorterDuffColorFilter(color, PorterDuff.Mode.SRC_ATOP));
-                Log.d("React-native-charts-wrapper", "drew bitmap " + color);
```
Comment (drhops) [inline android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java:115]: too noisy?  better to comment out if this matches a line in the wuxudong repo?

**Comment by drhops** [comment]
LGTM :pen:

File: android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java [hunk 4]
```diff
@@ -28,13 +30,24 @@ public RNOnChartValueSelectedListener(Chart chart) {
...
-            Chart chart = mWeakChart.get();
+            final Chart chart = mWeakChart.get();
 
             ReactContext reactContext = (ReactContext) chart.getContext();
             reactContext.getJSModule(RCTEventEmitter.class).receiveEvent(
                     chart.getId(),
                     "topSelect",
                     EntryToWritableMapUtils.convertEntryToWritableMap(entry));
+            
+            // Hack: unhighlight all values to trigger callback on re-taps
```
Comment (idealllee) [inline android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: wouldn't it trigger `onNothingSelected` on de-select? I could be misunderstanding it.

File: android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java (hunk 3 above)
Comment (jlo1) [reply android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java:115]: added this line of comment on my own PR here https://github.com/indiemoney/react-native-charts-wrapper/pull/5

File: android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java (hunk 4 above)
Comment (jlo1) [reply android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: right, the desired behavior for our UX though is that if you click on retirement icon, then close the edit-tray, and click on the retirement icon again, we want to still trigger the `onSelect` callback. (vs. in original code, the second click on retirement icon will trigger `onNothingSelected` and will have no indication on which icon was deselected)

Comment (idealllee) [reply android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: i see. why not call `chart.highlightValue(null, false);` when we close the edit-tray?

File: android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java (hunk 3 above)
Comment (drhops) [reply android/src/main/java/com/github/wuxudong/rncharts/data/DataExtract.java:115]: :ok_hand:

File: android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java (hunk 4 above)
Comment (jlo1) [reply android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: makes sense! will update! 🙇‍♀️

Comment (jlo1) [reply android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java]: Thanks for the suggestions @idealllee!
Exposed `resetHighlightValue` instead, and now just calling `setNativeProps` to update this whenever we get the `onSelect` event on a valid data point (see commit https://github.com/indiemoney/snowstorm/pull/776/commits/da6787d897abe74b9a27abf22dd025e298e27dc0)

**Comment by idealllee** [comment]
LGTM :high_brightness:

**Pull request merged**